mmengine/structures/base_data_element.py

```diff
@@ -309,7 +308,15 @@ def keys(self) -> list:
         Returns:
             list: Contains all keys in data_fields.
         """
-        return list(self._data_fields)
+        # We assume that the name of the attribute related to property is
+        # '_' + the name of the property. We use this rule to filter out
+        # private keys.
```

Comment:
Suggest adding a comment `# TODO: Use a more elegant way to solve this problem`.